browser/extensions/brave_component_loader.cc

```diff
@@ -89,12 +100,8 @@ void BraveComponentLoader::AddDefaultComponentExtensions(
   }
 
 #if BUILDFLAG(BRAVE_REWARDS_ENABLED)
-  if (!command_line.HasSwitch(switches::kDisableBraveRewardsExtension)) {
-    base::FilePath brave_rewards_path(FILE_PATH_LITERAL(""));
-    brave_rewards_path =
-        brave_rewards_path.Append(FILE_PATH_LITERAL("brave_rewards"));
-    Add(IDR_BRAVE_REWARDS, brave_rewards_path);
-  }
+  // Enable rewards extension if already opted-in
+  HandleRewardsEnabledStatus();
```

Comment:
nit `OnBraveRewardsEnabled` would be more conventional here

Reply:
@bridiver even if that function handles both enabled and disabled?